lib/CMakeLists.txt

```diff
@@ -1,3 +1,9 @@
+# TODO: Fix the fine grained tabegen dependencies
+# In the absence of fine grained tablegen dependencies we need to ensure that
+# Swift's libraries all build after the LLVM & Clang tablegen-generated headers
+# are generated.
+list(APPEND LLVM_COMMON_DEPENDS intrinsics_gen clang-tablegen-targets)
```

Comment:
Can you add a note on how this ensures that this works in in-tree builds work and maybe provide some information about what LLVM_COMMON_DEPENDs is in the LLVM build system?

Reply:
Agreed! `LLVM_COMMON_DEPENDS` is a global list of targets that is added as a dependency for every executable and library defined via `add_llvm_executable` and `add_llvm_library` (and, transitively, `add_swift_executable` and `add_swift_library`), right?